xlators/cluster/afr/src/afr-open.c

```diff
@@ -311,43 +442,19 @@ afr_fix_open(fd_t *fd, xlator_t *this)
     ret = loc_path(&local->loc, NULL);
     if (ret < 0)
         goto out;
-
     local->fd = fd_ref(fd);
     local->fd_ctx = fd_ctx;
+    local->strict_locks = _gf_false;
 
-    local->call_count = call_count;
-
-    gf_msg_debug(this->name, 0, "need open count: %d", call_count);
-
-    for (i = 0; i < priv->child_count; i++) {
-        if (!need_open[i])
-            continue;
-
-        if (IA_IFDIR == fd->inode->ia_type) {
-            gf_msg_debug(this->name, 0, "opening fd for dir %s on subvolume %s",
-                         local->loc.path, priv->children[i]->name);
-
-            STACK_WIND_COOKIE(frame, afr_openfd_fix_open_cbk, (void *)(long)i,
-                              priv->children[i],
-                              priv->children[i]->fops->opendir, &local->loc,
-                              local->fd, NULL);
-        } else {
-            gf_msg_debug(this->name, 0,
-                         "opening fd for file %s on subvolume %s",
-                         local->loc.path, priv->children[i]->name);
-
-            STACK_WIND_COOKIE(frame, afr_openfd_fix_open_cbk, (void *)(long)i,
-                              priv->children[i], priv->children[i]->fops->open,
-                              &local->loc, fd_ctx->flags & (~O_TRUNC),
-                              local->fd, NULL);
-        }
-
-        if (!--call_count)
-            break;
+    ret = synctask_new(this->ctx->env, __afr_fix_open, __afr_fix_open_cbk,
```

Comment:
Nit: double underscores are typically for functions under lock. IIRC, new coding guidelines don't recommend that also.

Reply:
Removed the double underscores and changed the function names.